docs/providers.md

```diff
@@ -3,10 +3,12 @@
 Providers retrieve data from your persistence layer.
 
 <!-- TOC -->
-* [Default providers](#default-providers)
-  * [Custom repository method](#custom-repository-method)
-  * [Custom repository arguments](#custom-repository-arguments)
-* [Custom providers](#custom-providers)
+* [Providers](#providers)
```

Comment:
Adding h1 header of the current page to the TOC looks a bit strange and doesn't match other doc files -- is it necessary? 

![image](https://github.com/user-attachments/assets/b8bd63a1-e9fe-465b-a55c-a9752c96e2e8)

![image](https://github.com/user-attachments/assets/635795db-ed6f-44b7-8207-122b8ce6e548)